core/src/main/kotlin/de/fraunhofer/iem/kpiCalculator/core/KpiCalculator.kt

```diff
@@ -10,97 +10,47 @@
 package de.fraunhofer.iem.kpiCalculator.core
 
 import de.fraunhofer.iem.kpiCalculator.core.hierarchy.KpiHierarchyNode
-import de.fraunhofer.iem.kpiCalculator.model.kpi.KpiId
+import de.fraunhofer.iem.kpiCalculator.core.hierarchy.KpiHierarchyNode.Companion.depthFirstTraversal
+import de.fraunhofer.iem.kpiCalculator.core.strategy.getKpiCalculationStrategy
 import de.fraunhofer.iem.kpiCalculator.model.kpi.KpiStrategyId
 import de.fraunhofer.iem.kpiCalculator.model.kpi.RawValueKpi
+import de.fraunhofer.iem.kpiCalculator.model.kpi.hierarchy.KpiCalculationResult
 import de.fraunhofer.iem.kpiCalculator.model.kpi.hierarchy.KpiHierarchy
-import de.fraunhofer.iem.kpiCalculator.model.kpi.hierarchy.KpiNode
 import de.fraunhofer.iem.kpiCalculator.model.kpi.hierarchy.KpiResultHierarchy
-
-private data class TreeUpdate(
-    val parent: KpiHierarchyNode,
-    val currentNode: KpiHierarchyNode,
-    val rawValueKpis: List<RawValueKpi>,
-)
+import io.github.oshai.kotlinlogging.KotlinLogging
 
 object KpiCalculator {
-    // XXX: Setup Logger
+
+    private val logger = KotlinLogging.logger {}
 
     fun calculateKpis(
         hierarchy: KpiHierarchy,
         rawValueKpis: List<RawValueKpi>,
+        strict: Boolean = false,
     ): KpiResultHierarchy {
-        val root = hierarchy.rootNode
-        val connectedHierarchyRoot = connectKpiHierarchyToRawValues(root, rawValueKpis)
-        depthFirstTraversal(connectedHierarchyRoot) { it.calculateKpi() }
-
-        return KpiResultHierarchy.create(KpiHierarchyNode.to(connectedHierarchyRoot))
-    }
-
-    private fun depthFirstTraversal(
-        node: KpiHierarchyNode,
-        seen: MutableSet<KpiHierarchyNode> = mutableSetOf(),
-        action: (node: KpiHierarchyNode) -> Unit,
-    ) {
-        if (seen.contains(node)) {
-            return
+        logger.info {
```

Comment:
Is this really worth logging in info level?
Also, i personally always put a `...` for Logs that have present progressive tense